services/generic/barrier_control.md

```diff
@@ -1,36 +1,94 @@
+| -    |                          |            |              |                                                                                                                                                                            |
+| in   | cmd.op.stop              | null       |              | Commence emergency stop of any operation.                                                                                                                                  |
+| -    |                          |            |              |                                                                                                                                                                            |
+| in   | cmd.state.get_report     | null       |              | Requests the current state of the device.                                                                                                                                  |
+| out  | evt.state.report         | string     | `stopped_at` | Reports the current state of the device, one of values defined in [`sup_states`](#service-properties) property.                                                            |
+| -    |                          |            |              |                                                                                                                                                                            |
+| in   | cmd.tstate.set           | string     |              | Sets the target state of the device to the one of values defined in [`sup_tstates`](#service-properties) property.                                                         |
+
+## Interface properties
+
+| Name         | Required | Example | Description                                                                                                                    |
+|--------------|----------|---------|--------------------------------------------------------------------------------------------------------------------------------|
+| `stopped_at` | No       | `"30"`  | Position as percentage value at which the barrier stopped on emergency halt, where `1` is near closed while `99` is near open. |
+
+## Service properties
+
+| Name             | Type      | Example                                    | Description                                                                                        |
+|------------------|-----------|--------------------------------------------|----------------------------------------------------------------------------------------------------|
+| `sup_notiftypes` | str_array | `["audio", "visual"]`                      | List of supported notification types. Possible values are: `audio`, `visual`.                      |
+| `sup_states`     | str_array | `["open", "closed", "closing", "opening"]` | List of supported states. Possible values are:  `closed`, `closing`, `stopped`, `opening`, `open`. |
```

Comment:
Possible -> Allowed sounds more natural here.

Reply:
List of supported target states is:

Reply:
Okies, I think we can treat this list as closed.